demo/style_configurator.c

```diff
@@ -0,0 +1,1048 @@
+
+// TODO design decisions
+// plural or not?  ie style_button or style_buttons?
+// use the duplicate array method, or just let the user
+// manually set those after calling the function by accessing ctx->style->*?
+
+static const char* symbols[NK_SYMBOL_MAX] =
+{
+    "NONE",
+    "X",
+    "UNDERSCORE",
+    "CIRCLE_SOLID",
+    "CIRCLE_OUTLINE",
+    "RECT_SOLID",
+    "RECT_OUTLINE",
+    "TRIANGLE_UP",
+    "TRIANGLE_DOWN",
+    "TRIANGLE_LEFT",
+    "TRIANGLE_RIGHT",
+    "PLUS",
+    "MINUS"
+};
+
+
+
+static void
+style_rgb(struct nk_context* ctx, const char* name, struct nk_color* color)
+{
+	nk_label(ctx, name, NK_TEXT_LEFT);
+	if (nk_combo_begin_color(ctx, *color, nk_vec2(nk_widget_width(ctx), 400))) {
+		nk_layout_row_dynamic(ctx, 120, 1);
+		struct nk_colorf colorf = nk_color_picker(ctx, nk_color_cf(*color), NK_RGB);
+		nk_layout_row_dynamic(ctx, 25, 1);
+		colorf.r = nk_propertyf(ctx, "#R:", 0, colorf.r, 1.0f, 0.01f,0.005f);
+		colorf.g = nk_propertyf(ctx, "#G:", 0, colorf.g, 1.0f, 0.01f,0.005f);
+		colorf.b = nk_propertyf(ctx, "#B:", 0, colorf.b, 1.0f, 0.01f,0.005f);
+
+		*color = nk_rgb_cf(colorf);
+
+		nk_combo_end(ctx);
+	}
+}
+
+// TODO style_style_item?  how to handle images if at all?
+static void
+style_item_color(struct nk_context* ctx, const char* name, struct nk_style_item* item)
+{
+	style_rgb(ctx, name, &item->data.color);
+}
+
+static void
+style_vec2(struct nk_context* ctx, const char* name, struct nk_vec2* vec)
+{
+	char buffer[64];
+	nk_label(ctx, name, NK_TEXT_LEFT);
+	sprintf(buffer, "%.2f, %.2f", vec->x, vec->y);
+	if (nk_combo_begin_label(ctx, buffer, nk_vec2(200,200))) {
+		nk_layout_row_dynamic(ctx, 25, 1);
+		nk_property_float(ctx, "#X:", -100.0f, &vec->x, 100.0f, 1,0.5f);
+		nk_property_float(ctx, "#Y:", -100.0f, &vec->y, 100.0f, 1,0.5f);
+		nk_combo_end(ctx);
+	}
+}
+
+// style_general? pass array in instead of static?
+static void
+style_colors(struct nk_context* ctx, struct nk_color color_table[NK_COLOR_COUNT])
```

Comment:
I don't think this function makes sense. Either this function have to be the only way to configure colors, or this function needs to be rethought. Currently, it will override changes made to color in other tabs when called, which I'm assuming is not what is intended.
![out](https://user-images.githubusercontent.com/3759175/76240364-6674b480-6233-11ea-8bfb-c92498be86cc.gif)

Reply:
They're kind of mutually exclusive though.  The Colors tab applies to all styles, so Colors->COLOR_TEXT is far more than Text->Color.  There's no logical way to connect the latter to the former.  The best (and I think most reasonable thing) I could do is maybe add a return value to style_rgb and not actually apply the color table unless one of the color drop downs was clicked (even if they didn't change it).

Reply:
I could rename the tab (and maybe the function) to be something like "Global Colors" or "Color Table"?

Reply:
Yeah, that could be a reasonable compromise.

Reply:
Obviously the general use case is do global color styling first and then tweak individual colors as needed in the other tabs.